web3.js/src/connection.ts

```diff
@@ -1720,6 +1720,15 @@ export type KeyedAccountInfo = {
   accountInfo: AccountInfo<Buffer>;
 };
 
+/**
+ * Callback function for subscribe error notifications
+ */
+export type SubscribeErrorCallback = (
```

Comment:
Some high level feedback:
1. It would be nice to have proper typing for the rpc args. I think an error info object could actually be the union of each subscription type plus its info fields rather than the built rpc args. Then developers can rely on the type system to pull info about the failed subscription out.
2. It'd be nice to return the subscription id here too for easier matching